app/src/main/java/com/android/calendar/agenda/AgendaAdapter.java

```diff
@@ -178,6 +179,12 @@ public void bindView(View view, Context context, Cursor cursor) {
         // When
         long begin = cursor.getLong(AgendaWindowAdapter.INDEX_BEGIN);
         long end = cursor.getLong(AgendaWindowAdapter.INDEX_END);
+        if (begin == 0) {
```

Comment:
can you give some explanation here?

Reply:
I use end time as reference. se task has begin = 0. if it's equal to 0 I calculate subtracting 30 mins

Reply:
what happens if an event has begin = 0?

Reply:
could we use `item.isTask` here to be sure to not modify other events?

Reply:
yes we can, but I think that you cannot have an event without start date.

Reply:
but an event can have 0 as a start date, or?

Reply:
an event (not task) has always a start date, also if it's a full day event

Reply:
sure, but `start` is a `long` so 0 is a valid start date.

Reply:
yes agree, but when you create an normal event 0 is impossibile to be set, so it's reserved for tasks

Reply:
why would it be impossible?

Reply:
cause when you create an event start date is always populated

Reply:
right, and I can populate it with 0.

Reply:
how? if you have a calendar picker

Reply:
and if u will be able to populate with 0 u will have an event that starts 30 mins before your end date and not an event that fullfill all the calendar

Reply:
by syncing it from CalDAV, for example or by going back to the Epoch.

Reply:
if u will be able to populate with 0 u will have an event that starts 30 mins before your end date and not an event that fullfill all the calendar

Reply:
so why not check `item.isTask`?

Reply:
why wrong? I think is better that abn event that starts from day 0 and fullfill all the calendar. however we can add the check "isTask" to prevent this

Reply:
because people may have valid reasons to do that. Please don't assume things.

Reply:
ok, I'll try a way to add isTAsk on AgendaAdapter

Reply:
fixed